packages/datadog-instrumentations/src/helpers/router-helper.js

```diff
@@ -0,0 +1,176 @@
+
+  return String(path)
+}
+
+// Flatten any route definition into an array of normalized path strings.
+function normalizeRoutePaths (path) {
+  const queue = Array.isArray(path) ? [...path] : [path]
+  const paths = []
+
+  while (queue.length) {
+    const current = queue.shift()
+
+    if (Array.isArray(current)) {
+      queue.unshift(...current)
+      continue
+    }
+
+    const normalized = normalizeRoutePath(current)
+
+    if (normalized !== null) {
+      paths.push(normalized)
+    }
+  }
+
+  return paths
+}
+
+function setRouterMountPath (router, mountPath) {
+  if (!router || typeof mountPath !== 'string') return
+  const existing = routerMountPaths.get(router)
+  if (existing) {
+    existing.add(mountPath)
+  } else {
+    routerMountPaths.set(router, new Set([mountPath]))
+  }
+}
+
+function getRouterMountPaths (router) {
+  const paths = routerMountPaths.get(router)
+  if (!paths) return []
+  return [...paths]
+}
+
+function setLayerMatchers (layer, matchers) {
+  layerMatchers.set(layer, matchers)
+}
+
+function getLayerMatchers (layer) {
+  return layerMatchers.get(layer)
+}
+
+function normalizeMethodName (method) {
+  return method === '_all' || method === 'all' ? '*' : method
+}
+
+function getRouteFullPaths (route, prefix) {
+  if (!route) return []
+
+  const routePaths = normalizeRoutePaths(route.path)
+  const pathsToPublish = routePaths.length ? routePaths : ['']
+
+  return pathsToPublish.map(routePath => joinPath(prefix, routePath))
+}
+
+function markAppMounted (router) {
+  if (router) appMountedRouters.add(router)
+}
+
+function isAppMounted (router) {
+  return appMountedRouters.has(router)
+}
+
+/**
+ * Normalise the optional mount argument provided to app.use()/router.use().
+ * Express accepts strings, regex, arrays (possibly nested), or
+ * no mount path at all; this helper returns the flattened set of paths along
+ * with the index where actual middleware arguments start.
+ */
+function extractMountPaths (args) {
+  const firstArg = args[0]
+  const hasMount = typeof firstArg === 'string' || firstArg instanceof RegExp || Array.isArray(firstArg)
+
+  if (!hasMount) {
+    return { mountPaths: ['/'], startIdx: 0 }
+  }
+
+  const paths = normalizeRoutePaths(firstArg)
+  return {
+    mountPaths: paths.length ? paths : ['/'],
+    startIdx: 1
+  }
+}
+
+/**
+ * Detect cycle router graphs.
+ */
+function hasRouterCycle (router, stack = new Set()) {
+  if (!router?.stack?.length) return false
+  if (stack.has(router)) return true
+
+  stack.add(router)
+
+  for (const layer of router.stack) {
+    if (!layer?.route && layer?.handle?.stack?.length) {
+      const hasCycle = hasRouterCycle(layer.handle, stack)
+      if (hasCycle) {
+        stack.delete(router)
+        return true
+      }
+    }
+  }
+
+  stack.delete(router)
+  return false
+}
+
+function wrapRouteMethodsAndPublish (route, paths, publish) {
+  if (!route || !paths?.length || typeof publish !== 'function') return
```

Comment:
`wrapRouteMethodsAndPublish` is not an exposed API and I don't think there's anywhere where we call it without a `publish` function. Same with `paths`: As far as I can see we always call with the a `paths` object. So why not simply:

```suggestion
  if (!route || !paths.length) return
```